crates/transaction-pool/src/batcher.rs

```diff
@@ -0,0 +1,273 @@
+//! Transaction batching for `Pool` insertion for high-throughput scenarios
+//!
+//! This module provides transaction batching logic to reduce lock contention when processing
+//! many concurrent transaction pool insertions.
+
+use crate::{AddedTransactionOutcome, PoolTransaction, TransactionOrigin, TransactionPool};
+use alloy_primitives::B256;
+use pin_project::pin_project;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::sync::{mpsc, oneshot};
+
+/// Configuration for tx pool batch insertion
+#[derive(Debug, Clone)]
+pub struct BatchTxConfig {
+    /// Channel buffer size for incoming batch tx requests
+    pub channel_buffer_size: usize,
+    /// Maximum number of transactions to batch insert at a time
+    pub max_batch_size: usize,
+}
+
+impl Default for BatchTxConfig {
+    fn default() -> Self {
+        Self { channel_buffer_size: 5000, max_batch_size: 3000 }
+    }
+}
+
+/// Error type for transaction batching operations
+#[derive(Debug, thiserror::Error)]
+pub enum BatchTxError {
+    /// Transaction batcher channel closed
+    #[error("Transaction batcher channel closed")]
+    BatcherChannelClosed,
+    /// Transaction response channel closed
+    #[error("Transaction response channel closed")]
+    ResponseChannelClosed,
+}
+
+/// A single batch transaction request
+/// All transactions processed through the batcher are considered local
+/// transactions (`TransactionOrigin::Local`) when inserted into the pool.
+#[derive(Debug)]
+pub struct BatchTxRequest<T: PoolTransaction> {
+    /// Tx to be inserted in to the pool
+    pool_tx: T,
+    /// Channel to send result back to caller
+    response_tx: oneshot::Sender<Result<B256, BatchTxError>>,
```

Comment:
we kinda need two layers here one for the actual Resul<AddedTransactionOutcome, PoolError> and then another outer layer for the case the channel closed, which we could avoid if we copy the hash so we can manually create a poolerror, but this feels a bit uneccesary, so I think we should use `Result<Result` or introduce another error type that unifies BatchTxError and PoolErr